test/testobjects/testutils.ts

```diff
@@ -98,3 +103,37 @@ export class TestUtils {
         return JSON.parse(dataJson) as IOrchestratorState;
     }
 }
+
+export class DummyHttpRequest implements HttpRequest {
+    public method: HttpMethod | null;
+    public user: HttpRequestUser | null;
+    public form: Form | null;
+
+    constructor(
+        public url = defaultRequestUrl,

```

Comment:
Is this new class just to simplify the logic, or does it also perform a change needed by the types upgrade?

Reply:
Both. Initially, the `HttpRequest` type was simple enough that it could be constructed on the fly when necessary, with something like:

```TS
 const requestObj: HttpRequest = {
                method: "GET",
                url: defaultRequestUrl,
                headers: {},
                query: {},
                params: {},
            };
```

But now, the `HttpRequest` type also includes methods such as `parseFormBody()`, `get()`, etc., such that it would be untenable to construct a brand new instance compliant with the type each time